src/compute/src/compute_state.rs

```diff
@@ -432,8 +432,11 @@ impl<'a, A: Allocate> ActiveComputeState<'a, A> {
     pub fn report_compute_frontiers(&mut self) {
         let mut new_uppers = Vec::new();
 
+        // Maintain a single allocation for `new_frontier` to avoid allocating on every iteration.
+        let mut new_frontier = Antichain::new();
+
         for (&id, collection) in self.compute_state.collections.iter_mut() {
-            let mut new_frontier = Antichain::new();
+            new_frontier.clear();
```

Comment:
I don't think clearing is necessary here, based on that before #20090 it also didn't happen. Presumably both `read_upper` and `clone_from` do their own clears.

While writing this I have convinced myself that this is still a good thing to do, as a matter of defensive programming. So feel free to disregard this :)

Reply:
Yup, you're correct, but I added it to not confuse myself again that there is any important state maintained across loop iterations.